virtool/groups/api.py

```diff
@@ -48,17 +48,17 @@ async def post(
             201: Successful operation
             400: Group already exists
         """
-        group_id = data.group_id
+        name = data.name
 
         try:
-            group = await get_data_from_req(self.request).groups.create(group_id)
+            group = await get_data_from_req(self.request).groups.create(name)
         except ResourceConflictError:
             raise HTTPBadRequest(text="Group already exists")
 
         return json_response(
             GroupResponse.parse_obj(group),
             status=201,
-            headers={"Location": f"/groups/{group.id}"},
+            headers={"Location": f"/groups/{group.name}"},
```

Comment:
Location still uses the `id`.
```suggestion
            headers={"Location": f"/groups/{group.id}"},
```